inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.h

```diff
@@ -29,27 +32,44 @@ class MKLDNNPadNode : public MKLDNNNode {
         SYMMETRIC = 3
     };
 
-    void padConstantOrEdge(const float *srcData, float* dstData,
-                           InferenceEngine::SizeVector srcDims, InferenceEngine::SizeVector dstDims,
-                           const bool isEdge = false);
-    void padReflectOrSymmetric(const float *srcData, float* dstData,
-                               InferenceEngine::SizeVector srcDims, InferenceEngine::SizeVector dstDims,
-                               const bool isSymmetric = false);
+    void padConstant();
+    template<typename T> void padConstantCommon();
+    void padConstantZero();
+    void padEdge();
+    void padReflectOrSymmetric(const bool isSymmetric = false);
+
+    void prepareParams();
 
-    size_t getWorkAmountDst() const;
+    inline void getDstIdx(const InferenceEngine::SizeVector& indexes, size_t& dstIdx) const;
```

Comment:
This fuctions although could be moved to the cpp as `static inline`.

Reply:
Discussed offline. This method accesses class fields